File: app/src/main/java/com/benoitquenaudon/tvfoot/red/app/domain/matches/displayable/BroadcasterRowDisplayable.java
package com.benoitquenaudon.tvfoot.red.app.domain.matches.displayable;

import android.os.Parcelable;
import com.google.auto.value.AutoValue;

@AutoValue public abstract class BroadcasterRowDisplayable implements Parcelable {
  public abstract String name();

  public abstract String logoPath();

  public static BroadcasterRowDisplayable create(String name, String broadcasterCode) {
    return new AutoValue_BroadcasterRowDisplayable(name,
        String.format("/images/broadcasters/%s_120.png", broadcasterCode));
  }
}

File: app/src/main/java/com/benoitquenaudon/tvfoot/red/app/domain/matches/displayable/BroadcasterRowDisplayable.kt
package com.benoitquenaudon.tvfoot.red.app.domain.matches.displayable

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize

@Parcelize
data class BroadcasterRowDisplayable(
    val name: String,
    val broadcasterCode: String
) : Parcelable {
  val logoPath: String
    get() = "/images/broadcasters/${broadcasterCode}_120.png"
}
